src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java

```diff
@@ -823,6 +823,36 @@ public void testLoggedTime() throws IOException {
 
     }
 
+    @Test
+    public void testLoggedTimeInProcessFiles() throws IOException {
+        final CheckstyleAntTaskLogStub antTask = new CheckstyleAntTaskLogStub();
+        antTask.setConfig(getPath(CONFIG_FILE));
+        antTask.setProject(new Project());
+        antTask.setFile(new File(getPath(FLAWLESS_INPUT)));
+        final long startTime = System.currentTimeMillis();
+        antTask.execute();
+        final long endTime = System.currentTimeMillis();
+        final long overallLoggedTime = endTime - startTime;
+        final List<MessageLevelPair> loggedMessages = antTask.getLoggedMessages();
+
+        assertLoggedTime(loggedMessages, overallLoggedTime, "locate");
+        assertLoggedTime(loggedMessages, overallLoggedTime, "process");
+    }
+
+    private static void assertLoggedTime(List<MessageLevelPair> loggedMessages,
+                                         long overallLoggedTime, String action) {
+        loggedMessages.stream()
+            .filter(msg -> msg.getMsg().startsWith("To " + action + " the files took"))
+            .findFirst()
+            .ifPresentOrElse(msg -> {
+                assertWithMessage("Logged time to " + action
+                                      + " the files should be less than the overall logged time")
+                    .that(getNumberFromLine(msg.getMsg()))
+                    .isAtMost(overallLoggedTime);
+            }, assertWithMessage(
+                "Log message with 'To " + action + " the files took' was not found")::fail);
+    }
```

Comment:
I think it's better to make this method more generic:
```java
assertLoggedTime(loggedMessages, overallLoggedTime, "To locate the files took");
assertLoggedTime(loggedMessages, overallLoggedTime, "To process the files took");

private static void assertLoggedTime(List<MessageLevelPair> loggedMessages,
                                     long overallLoggedTime, String expectedMessage) {
    loggedMessages.stream()
        .filter(msg -> msg.getMsg().startsWith(expectedMessage))
        .findFirst()
        .ifPresentOrElse(msg -> {
            assertWithMessage("Logged time in '" + expectedMessage "' "
            	              + "must be less than the overall logged time")
                .that(getNumberFromLine(msg.getMsg()))
                .isAtMost(overallLoggedTime);
        }, assertWithMessage("Log message expected: " + expectedMessage)::fail);
}
```
and we probably can reuse this method in `testLoggedTime()` too

Reply:
Done